pkg/consuldp/lifecycle.go

```diff
@@ -211,3 +207,81 @@ func (m *lifecycleConfig) gracefulShutdown() {
 	// Wait for context timeout to elapse
 	wg.Wait()
 }
+
+func (m *lifecycleConfig) gracefulStartupHandler(rw http.ResponseWriter, _ *http.Request) {
+
+	//Unlike in gracefulShutdown, we want to delay the OK response until envoy is ready
+	//in order to block application container.
```

Comment:
This would be a great place to emit a log. "Request received to graceful startup. Returning when Dataplane is in a ready state."